assets/src/components/Sidebar/CampaignSidebar.js

```diff
@@ -123,14 +139,17 @@ export class CampaignSidebar extends Component {
         </PluginSidebarMoreMenuItem>
         <PluginSidebar
           name={ CampaignSidebar.getId() }
-          title={ __( 'Campaign Options', 'planet4-blocks-backend' ) }
+          title={ __('Campaign Options', 'planet4-blocks-backend') }
         >
-          { parent ? <PostParentLink parent={ parent }/> :
-            <LegacyThemeSettings
-              theme={theme}
-              handleThemeSwitch={ this.handleThemeSwitch }
-            />
-          }
+          { !!parent && <PostParentLink parent={ parent }/> }
+          { !parent && meta && <NewThemeSettings currentTheme={meta.theme} onChange={ async value => {
```

Comment:
As a best practice and thinking about the performance, it's better always to move out all`lambda` from the `render` render. Take in mind that with every render you are creating a new instance of this function.

```
const onChangeHandler = useCallback((value) => {
  this.handleThemeSwitch('theme', value, meta);
}, [
meta,
]);
```

Reply:
`useCallback` is only better for performance if the component is rendered extremely frequently (maybe in some edge cases too). If the component is not rendered frequently, its performance is actually (ever so slightly) worse. See https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
I agree with you. Feel free to avoid using `useCallback` if this component isn't extremely frequently!